core/src/main/scala/org/apache/spark/network/netty/NettyBlockTransferService.scala

```diff
@@ -73,9 +73,13 @@ class NettyBlockTransferService(conf: SparkConf, securityManager: SecurityManage
       val server = transportContext.createServer(port, bootstraps.asJava)
       (server, server.getPort)
     }
-
     val portToTry = conf.getInt("spark.blockManager.port", 0)
-    Utils.startServiceOnPort(portToTry, startService, conf, getClass.getName)._1
+
+    if(Utils.hasPortRange(conf)){
+      Utils.startServiceFromPortRange(portToTry, startService, conf, getClass.getName)._1
```

Comment:
We only affect netty layer if the Utils.hasPortRange(conf) is true which is only set at MesosExecutorBackend there is no other way to be set so we are safe. The only way to pass info down is through conf otherwise the code base needs much bigger refactoring.